### PR TITLE
take advantage of DigitalOcean resources

### PR DESCRIPTION
**Pull request by pachadotdev**
this PR adds:

- DO Rstudio image as default
- DO CRAN mirror as default
- Ubuntu as default (better aligned with DO tools for monitoring droplets)
- Functions to create users
- Snapshotting guidelines
- Exhaustive tests (but still doesn't cover all functions)

I'm happy to do changes to this PR if required

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -215,6 +211,13 @@ export(tag_rename)
 export(tag_resource)
 export(tag_resource_delete)
 export(tags)
+export(ubuntu_add_swap)
```
Comment (sckott) [inline NAMESPACE:214]: Can you talk about why the change from debian to ubuntu in these function names? It might make sense, I don't know. If they are changed, we need to make sure when the old function names are used they give a useful message pointing to the new functions. E.g., this guidance https://devguide.ropensci.org/evolution.html#functions-changing-function-names

File: R/droplet-actions.R [hunk 2]
```diff
@@ -84,7 +84,7 @@
 #' }
 droplet_create <- function(name = random_name(),
                         size = getOption("do_size", "s-1vcpu-1gb"),
-                        image = getOption("do_image", "ubuntu-18-04-x64"),
+                        image = getOption("do_image", "rstudio-20-04"),
```
Comment (sckott) [inline R/droplet-actions.R:87]: What is `rstudio-20-04`? Is it just a blank ubuntu machine? Or have rstudio server installed? If the latter, I lean towards leaving it as a plain ubuntu machine - and perhaps creating a new function, e.g, `rstudio_create` if you want a similar functionality specifically for rstudio server. Thoughts?

File: R/droplets_create.R [hunk 3]
```diff
@@ -34,8 +34,8 @@
 #' }
 droplets_create <- function(names = NULL,
                            size = getOption("do_size", "s-1vcpu-1gb"),
-                           image = getOption("do_image", "ubuntu-18-04-x64"),
-                           region = getOption("do_region", "sfo3"),
+                           image = getOption("do_image", "rstudio-20-04"),
+                           region = getOption("do_region", "tor1"),
```
Comment (sckott) [inline R/droplets_create.R:38]: Same question here about `rstudio` image vs. `ubuntu`

File: .github/workflows/R-CMD-check.yml [hunk 4]
```diff
@@ -12,16 +12,19 @@ jobs:
...
-        - { os: ubuntu-16.04, r: 'release', rspm: "https://packagemanager.rstudio.com/cran/__linux__/xenial/latest"}
+        - { os: ubuntu-20.04, r: '4.0', args: "--no-manual" }
 
     env:
       R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
-      RSPM: ${{ matrix.config.rspm }}
 
     steps:
+      - name: Install SSH key
+        uses: shimataro/ssh-key-action@v2.3.0
```
Comment (sckott) [inline .github/workflows/R-CMD-check.yml:22]: CI broken because the key is not available in secrets in this repo. Do you plan to add the ssh key and known hosts information here after merging?

**Review by sckott — CHANGES_REQUESTED** [review]
Thanks! Some questions and comments

File: DESCRIPTION [hunk 5]
```diff
@@ -14,9 +14,11 @@ Authors@R: c(
     person("Bob", "Rudis", role = "ctb", email = "bob@rudis.net"),
     person("Bryce", "Mecum", role = "ctb", email = "brycemecum@gmail.com",
            comment = c("ORCID" = "0000-0002-0381-3766")),
+    person("Mauricio", "Vargas", role = "ctb", email = "mavargas11@uc.cl",
+           comment = c(ORCID = "0000-0003-1017-7574")),
     person("RStudio", role = "cph")
     )
-License: MIT + file LICENSE
+License: Apache License (>= 2)
```
[the remark this answers is not in the document]
Comment (pachadotdev) [reply DESCRIPTION:21]: based on what I do for work (try to improve the Arrow R package), I discovered something called 'the Apache way' (https://apache.org/theapacheway/), and using the Apache license is aligned with it with the advantage that it prevents patent trolls actions

[the remark this answers is not in the document]
Comment (pachadotdev) [reply DESCRIPTION:21]: Apache also happens to be the license for most DigitalOcean OSS products

File: NAMESPACE (hunk 1 above)
Comment (pachadotdev) [reply NAMESPACE:214]: I used a lot of DO products for the thesis, and I even have a DO email since the thesis involved creating a few things such as a CRAN mirror and a dedicated R image. Ubuntu works super well with DO's agent-metrics, a service that integrates well with their cloud dashboard and allows servers monitoring

I'm gonna add dummy functions with deprecation messages, but DO + Ubuntu is a very nice combo

File: R/droplet-actions.R (hunk 2 above)
Comment (pachadotdev) [reply R/droplet-actions.R:87]: it's this image https://marketplace.digitalocean.com/apps/rstudio
it features R + Tidyverse

counterproposal, use "ubuntu-20-04" as default argument, and mention in the docs that "rstudio" shall give you a full setup (Even with RStudio Server) in 30 secs

File: .github/workflows/R-CMD-check.yml (hunk 4 above)
Comment (pachadotdev) [reply .github/workflows/R-CMD-check.yml:22]: yes, I'll pass you the key and all the stuff, I asked DO people and we can run this project at zero cost (no abuse clause, of course) with that key, they liked the idea of extending tests to droplet creation, snapshotting ,etc.

File: DESCRIPTION (hunk 5 above)
[the remark this answers is not in the document]
Comment (sckott) [reply DESCRIPTION:21]: Okay, apache sounds good

File: .github/workflows/R-CMD-check.yml (hunk 4 above)
Comment (sckott) [reply .github/workflows/R-CMD-check.yml:22]: Sounds good

File: NAMESPACE (hunk 1 above)
Comment (sckott) [reply NAMESPACE:220]: Okay, makes sense

File: R/droplet-actions.R (hunk 2 above)
Comment (sckott) [reply R/droplet-actions.R]: Either way -up to you now! 

But changing to `rstudio-20-04` does change the behavior of the function quite a bit in that it's not just a plain ubuntu machine anymore. I guess at least make sure the change is documented if you keep this change :)

**Review by sckott — APPROVED** [review]
